flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/source/enumerator/cursor/StartCursor.java

```diff
@@ -73,18 +73,6 @@ static StartCursor fromMessageId(MessageId messageId, boolean inclusive) {
         return new MessageIdStartCursor(messageId, inclusive);
     }
 
-    /**
-     * This method is designed for seeking message from event time. But Pulsar didn't support
-     * seeking from message time, instead, it would seek the position from publish time. We only
-     * keep this method for backward compatible.
-     *
-     * @deprecated Use {@link #fromPublishTime(long)} instead.
-     */
-    @Deprecated
```

Comment:
I can approve this change into this PR, although it can be out of the scope. But you can write a release note for what deprecated methods get removed in 4.0.0 so users will be aware of it. @syhily 

I agree that we should remove these methods since they're not working deprecated methods but actually buggy methods.

Reply:
FYI @MartijnVisser

Reply:
I see. But I don't know where to add it? Since the Pulsar 4.0.0 will introduce a lot of break changes. Maybe I can add these changes somewhere in https://cwiki.apache.org/confluence/display/FLINK/1.17+Release?

Reply:
If Pulsar 4.0.0 will introduce breaking changes, these should be included in the `release notes` field of the Jira ticket which belongs to the change that will introduce the breaking change. In that field, you should also include the new/alternative solution

Reply:
@MartijnVisser Can you post a link here which I can add a Pulsar 4.0.0 release note with all the things we have changed for now?